File: LeetCode/Easy/0303-range-sum-query-immutable/0303-range-sum-query-immutable.cs
public class NumArray {
    private readonly int[] prefix;

    public NumArray(int[] nums) {
        prefix = new int[nums.Length + 1];
        for (var i = 0; i < nums.Length; i++)
        {
            prefix[i + 1] = prefix[i] + nums[i];
        }
    }
    
    public int SumRange(int left, int right)
        => prefix[right + 1] - prefix[left];
}

/**
 * Your NumArray object will be instantiated and called as such:
 * NumArray obj = new NumArray(nums);
 * int param_1 = obj.SumRange(left,right);
 */

File: LeetCode/Easy/0303-range-sum-query-immutable/0303-range-sum-query-immutable.py
class NumArray:

    def __init__(self, nums: List[int]):
        self.prefixes = [0]
        
        running_sum = 0
        for n in nums:
            running_sum += n
            self.prefixes.append(running_sum)

    def sumRange(self, left: int, right: int) -> int:
        return self.prefixes[right + 1] - self.prefixes[left]


# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(left,right)